### PR TITLE
Make macOS binary cache deterministic. Workaround for #19623

### PR DESCRIPTION
**Pull request by autoantwort**
Workaround for #19623

### CODE REVIEW

**Comment by autoantwort** [comment]
paraview is currently generally broken, this is not caused by this PR

**Comment by JackBoosY** [comment]
I think we can't simply do that.

cc @strega-nil-ms

**Comment by cenit** [comment]
in any case, valid or not, please add a comment with a reference to the issue next to the code, otherwise in a month or two someone will reverse it without any knowledge, "to improve m1 compatibility"

**Comment by autoantwort** [comment]
Not needed anymore when https://github.com/microsoft/vcpkg-tool/pull/163 gets merged

File: scripts/bootstrap.sh [hunk 1]
```diff
@@ -299,7 +299,7 @@ echo "Building vcpkg-tool..."
 rm -rf "$baseBuildDir"
 mkdir -p "$buildDir"
 vcpkgExtractArchive "$tarballPath" "$srcBaseDir"
-cmakeConfigOptions="-DCMAKE_BUILD_TYPE=Release -G 'Ninja' -DCMAKE_MAKE_PROGRAM='$ninjaExe'"
+cmakeConfigOptions="-DCMAKE_BUILD_TYPE=Release -G 'Ninja' -DCMAKE_MAKE_PROGRAM='$ninjaExe' -DCMAKE_OSX_ARCHITECTURES='x86_64'"
```
Comment (strega-nil-ms) [inline scripts/bootstrap.sh:302]: I'd prefer that we choose the correct architecture here rather than always x86_64; I want an arm64 binary on my M1.

Comment (autoantwort) [reply scripts/bootstrap.sh:302]: Yeah but that maybe breaks the binary cache. This PR is unnecessary when https://github.com/microsoft/vcpkg-tool/pull/163 gets merged soon

Comment (strega-nil-ms) [reply scripts/bootstrap.sh:302]: Could you do `'-DCMAKE_OSX_ARCHITECTURES=x86_64;arm64'`?

Comment (strega-nil-ms) [reply scripts/bootstrap.sh:302]: Also, I don't understand how setting it to `arm64` breaks the binary cache any less or more than `x64`?

Comment (autoantwort) [reply scripts/bootstrap.sh:302]: > Also, I don't understand how setting it to arm64 breaks the binary cache any less or more than x64?

Yes that is true, setting it to arm64 also works. But some ports don't build when you build the tool as arm64, so x64 is more reliable. 

> Could you do '-DCMAKE_OSX_ARCHITECTURES=x86_64;arm64'?

I could. But we should merge microsoft/vcpkg-tool#163

**Comment by autoantwort** [comment]
Close in favor of microsoft/vcpkg-tool#163

**Pull request closed without merging**